# Serial 1

Simple analog receive over serial

In [40]:
import conda
import serial
import time

In [43]:
# open serial port
ser = serial.Serial('COM5', 9600, timeout = 1)

# wait for it to open
time.sleep(2)

# RECEIVE BYTES
# read a byte string
b = ser.readline()

# BYTES -> STRING
# decode byte string into Unicode 
string_n = b.decode()  

# STRING -> STRING with termination characters removed
# remove \n and \r
string = string_n.rstrip() 

# STRING -> NUMBER (float)
# convert string to float
flt = int(string)       
print(flt)

# close serial port
ser.close()

983


In [42]:
print(b) # b'83\r\n'
print(type(b))

print(string_n)
print(type(string_n))

print(string)
print(type(string))

print(flt)
print(type(flt))

b'983\r\n'
<class 'bytes'>
983

<class 'str'>
983
<class 'str'>
983
<class 'int'>


## RAZUMEVANJE

Zakaj je print v novo vrstico? Zato ker ta string še vsebuje termination character, ki jih moramo odstranit

<code>
b'983\r\n'
<class 'bytes'>
983

<class 'str'>
983
<class 'str'>
983
<class 'int'>
</code>